### ISSUE TITLE
Web Crypto API polyfill breaks Wasp's server when using Node.js 20

### ISSUE THREAD

**Issue by infomiho** [wasp-lang/wasp#1833]
> [!WARNING]
> Wasp doesn't work correctly with Node.js 20 because we try to load a polyfill which is only needed for Node.js 18 and this breaks Wasp's server when using Node.js 20

This was reported on Discord when one of our users tried to use Wasp 0.12 with Node.js 20:

```
File: /app/.wasp/out/server/src/polyfill.ts:7
 globalThis.crypto = webcrypto as Crypto;
 TypeError: Cannot set property crypto of #<Object> which has only a getter
     at <anonymous> (/Users/Dealwap/Desktop/resume-sass/app/.wasp/out/server/src/polyfill.ts:7:21)
```
https://ptb.discord.com/channels/686873244791210014/1212533517360177253/1212803377918902382

## Explanation 

We used Node.js 18 before and we ALWAYS needed the polyfill the Web Crypto API as per Lucia's docs recommendation: https://lucia-auth.com/getting-started/

### What can we do to fix it

Since the polyfill just defines a property on `globalThis` and the error message is about Node.js 20 complaining that we are setting a property which is only defined as a getter (which means _the property exists_) we could fix this by applying the polyfill _only_ if the `globalThis.crypto` property doesn't exist.

### Alternative solutions

We could detect the user's Node.js version and either import the polyfill or not?

## Future work

We need to introduce testing with Node.js 20 (or any supported versions) in our CI. https://github.com/wasp-lang/wasp/issues/1835

**Comment by Zeko369** [comment]
I know it costs more money, but many run CI against Node18 and 20?

**Comment by infomiho** [comment]
I believe our CI usage on Github is free since we are open source 😄 sooo the only cost would be longer CI times, we could maybe limit ourselves to running Node.js 20 tests on Linux only maybe? 🤔

**Comment by Zeko369** [comment]
It's not gonna be longer since matrix by default runs in parallel

Also CI is free for OSS, damn 🚀?

**Comment by Martinsos** [comment]
Yes it's free! I think there are some limits, if I remember correctly we get some amount of credits/something to spend. Let's create a separate issue for adding more node versions into CI. We can run those only on Linux, to keep it all together faster, and we can maybe even run them in parallel.

Crypto polyfill -> ok, solution with applying it only if property is not defined makes the most sense to me.
